### PR TITLE
node pools: implement `node pool init` command

### PR DESCRIPTION
**Pull request by tgross**
Implement a `nomad node pool init` command that generates an example spec file in either HCL or JSON format.

### CODE REVIEW

File: command/node_pool_init.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	"os"
+	"strings"
+
+	"github.com/posener/complete"
+)
+
+const (
+	// DefaultHclNodePoolInitName is the default name we use when initializing
+	// the example node pool spec file in HCL format
+	DefaultHclNodePoolInitName = "pool.np.hcl"
```
Comment (lgfa29) [inline command/node_pool_init.go:19]:
```suggestion
	DefaultHclNodePoolInitName = "pool.nomad.hcl"
```

I think text editors would prefer if we were consistent with file extension? Wrapping things in blocks then provides a consistent grammar.

File: command/node_pool_init.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+func (c *NodePoolInitCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *NodePoolInitCommand) Name() string { return "node pool init" }
+
+func (c *NodePoolInitCommand) Run(args []string) int {
+	var outFmt string
+	var quiet bool
```
Comment (lgfa29) [inline command/node_pool_init.go:67]: Was this supposed to be a flag?

File: command/node_pool_init.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  "Name": "example",
+  "Description": "Example node pool",
+  "Meta": {
+    "environment": "prod",
+    "owner": "sre"
+  },
+  "SchedulerConfiguration": {
+    "SchedulerAlgorithm": "spread"
+  }
+}
```
Comment (lgfa29) [inline command/node_pool_init.go:161]: Good candidates for `go:embed?` 😄 
https://github.com/hashicorp/nomad/blob/main/command/asset/asset.go

**Review by lgfa29 — APPROVED** [review]
Just added some bikeshedding 😄

File: command/node_pool_init.go (hunk 1 above)
Comment (tgross) [reply command/node_pool_init.go:19]: Makes sense. We didn't do that for variables (or volumes, or quotas, or namespaces... _sigh_ :grinning: ) but if we can do it going forward it'll let us define that grammar as a single "nomad grammar"

File: command/node_pool_init.go (hunk 2 above)
Comment (tgross) [reply command/node_pool_init.go:67]: :man_facepalming:

File: command/node_pool_init.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  # configuration as described at:
+  # https://developer.hashicorp.com/nomad/docs/commands/operator/scheduler/set-config
+  #
+  # Available only in Nomad Enterprise.
+  scheduler_configuration {
+
+    # scheduler_algorithm is the scheduling algorithm to use for the pool.
+    # If not defined, the global cluster scheduling algorithm is used.
+    scheduler_algorithm = "spread"
+  }
```
[the remark this answers is not in the document]
Comment (tgross) [reply command/node_pool_init.go:146]: Good call.

File: command/node_pool_init.go (hunk 3 above)
Comment (tgross) [reply command/node_pool_init.go]: Good call. Done.

File: website/content/docs/commands/node-pool/init.mdx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+nomad node pool init <filename>
+```
+
+When no filename is supplied, a default filename of "pool.np.hcl" or
+"pool.np.json" will be used depending on the output format.
+
+## Init Options
+
+- `-out` `(enum: hcl | json)`: Format of generated node pool
+  specification. Defaults to `hcl`.
```
Comment (lgfa29) [inline website/content/docs/commands/node-pool/init.mdx:25]: Missing the new `quiet` flag.

**Pull request merged**